extensions/virtual-threads/runtime/src/main/java/io/quarkus/virtual/threads/VirtualThreadsRecorder.java

```diff
@@ -100,45 +106,39 @@ static ExecutorService newVirtualThreadPerTaskExecutorWithName(String prefix)
         Method ofVirtual = Thread.class.getMethod("ofVirtual");
         Object vtb = ofVirtual.invoke(VirtualThreadsRecorder.class);
         Class<?> vtbClass = Class.forName("java.lang.Thread$Builder$OfVirtual");
-        Method name = vtbClass.getMethod("name", String.class, long.class);
-        vtb = name.invoke(vtb, prefix, 0);
+        // .name()
+        if (prefix != null) {
+            Method name = vtbClass.getMethod("name", String.class, long.class);
+            vtb = name.invoke(vtb, prefix, 0);
+        }
+        // .uncaughtExceptionHandler()
+        Method uncaughtHandler = vtbClass.getMethod("uncaughtExceptionHandler", Thread.UncaughtExceptionHandler.class);
+        vtb = uncaughtHandler.invoke(vtb, new Thread.UncaughtExceptionHandler() {
+            @Override
+            public void uncaughtException(Thread t, Throwable e) {
+                logger.errorf(e, "Thread %s threw an uncaught exception:", t);
```

Comment:
Should it be Virtual Thread?

Reply:
I kept the error message the same as the worker thread pool one. The thread toString has enough information such as `VirtualThread[#146]/runnable@ForkJoinPool-1-worker-1`